In [10]:
import numpy as np


def fred(a,b):
    c = (a+b)*12
    return c

a = 13.2
b = 3.9
c = fred(a,b)
print("c = ", c)



#rule1
def rule1(c, dA):
    dQ = np.abs(c)*dA
    return dQ

c = 1.0
A = .3
dA = .0005
dtotal = c * A
errdtotal = rule1(c, dA)
print("Rule1:dtotal =", dtotal)
print("Rule1:errdtotal =", errdtotal)


#rule2
def rule2(c, m, A, dA):
    dQ = np.abs(c*m*A**(m-1))*dA
    return dQ

c = 1
m = 2
A = 3
dA = .0005
dtotal = c*A**m
errdtotal = rule2(c, m, A, dA)
print("Rule2:dtotal =", dtotal)
print("Rule2:errdtotal =", errdtotal)


#rule3
def rule3(dA, dB):
    dQ = np.sqrt(dA**2+dB**2)
    return dQ

A = 1.0
B = 0.3
dA = .0005
dB = .0005
dtotal = A + B
errdtotal = rule3(dA, dB)
print("Rule3:dtotal =", dtotal)
print("Rule3:errdtotal =", errdtotal)


#rule4
def rule4(Q, m, dA, A, n, dB, B):
    dQ = np.abs(Q)*np.sqrt((m*(dA/A))**2+(n*(dB/B))**2)
    return dQ

Q = 2.2530
m = 1
dA = .0027
A = 1.0127
n = -.5
dB = .005
B = .99
c = 2.21
dtotal = c*(A**m)*(B**n)
errdtotal = rule4(Q, m, dA, A, n, dB, B)
print("Rule4:dtotal =", dtotal)
print("Rule4:errdtotal =", errdtotal)



#average from lab #6
x = np.array([1.1,1.3,1.4,0.9,0.95,1.05])
xmean = np.average(x)
xstdev = np.std(x)
print ("xmean =", xmean)
print ("xstdev =", xstdev)



c =  205.2
Rule1:dtotal = 0.3
Rule1:errdtotal = 0.0005
Rule2:dtotal = 9
Rule2:errdtotal = 0.003
Rule3:dtotal = 1.3
Rule3:errdtotal = 0.0007071067811865475
Rule4:dtotal = 2.249341968083739
Rule4:errdtotal = 0.008273512642662996
xmean = 1.1166666666666667
xstdev = 0.17950549357115014


The average is 1.117 and the standard deviation is .1795

$\delta Q = \sqrt{(\delta A)^2 + (\delta B)^2}$

$\delta V_i = V_i * \sqrt{(m * \frac{\delta X}{X})^2 + (n * \frac{\delta Y}{Y})^2}$

$\delta V_i = 2.2530 * \sqrt{(\frac{.0027}{1.0127})^2 + (\frac{-1}{2}*\frac{.005}{.99})^2} = .008 $
